store/src/java/com/zimbra/cs/service/UserServlet.java

```diff
@@ -382,8 +382,7 @@ public void doGet(HttpServletRequest req, HttpServletResponse resp) throws Servl
             checkTargetAccountStatus(context);
 
             if (!checkTwoFactorAuthentication(context)) {
-                resp.addHeader(AuthUtil.WWW_AUTHENTICATE_HEADER, getRealmHeader(req, null));
```

Comment:
I am not too familiar with the Zimbra codebase yet, but want to confirm if two-factor auth is mandatory? or is enabled based on flag/attribute?

Reply:
Two-Factor Auth can be either mandatory or optional (or even unavailable as a feature) to a given a account depending on the settings for its Class of Service. The `checkTwoFactorAuthentication` method on line 514 finds whether the account should be using 2FA while trying to access a URL with Basic authentication alone, for the purpose of throwing the ServiceException from line 385 in such scenario, as required in the user story.